firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLitePersistence.java

```diff
@@ -455,6 +459,122 @@ private Cursor startQuery() {
     }
   }
 
+  /**
+   * Encapsulates a query whose parameter list is so long that it might exceed SQLite limit.
+   *
+   * <p>SQLite limits maximum number of host parameters to 999 (see
+   * https://www.sqlite.org/limits.html). This class wraps most of the messy details of splitting a
+   * large query into several smaller ones.
+   */
+  static class LongQuery {
+    private final SQLitePersistence db;
+    private final String head;
+    private final String tail;
+    private final List<Object> argsHead;
+    private final List<Object> allArgs;
+
+    private int queriesPerformed = 0;
+    private Iterator<Object> argsIter;
+
+    private static final int LIMIT = 900;
```

Comment:
I'm concerned about off-by-one errors, but setting the limit so much below 999 is kinda superstitious... Let me know what you think.

Reply:
I'm on board with 900.